src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+/**
+ * Parser that takes user input and outputs commands DukeBot can understand.
+ */
+public class Parser {
+
+    /**
+     * Static method used to parse player inputs into a String array
+     * that is understood by Duke.
+     *
+     * @param s String input by user to be parsed.
+     * @return String array to be processed by Duke.
+     * @throws InvalidTaskException Thrown if user input wrong task format.
+     * @throws EmptyDescriptionException Throw if task description is empty.
+     * @throws InvalidFormatException Thrown if invalid task format.
+     */
+    public static String[] parseInput(String s) throws InvalidTaskException,
```

Comment:
Despite its relative length, you managed to avoid deep nesting of if-else statements and were able to structure the code neatly. Well done!